probe-rs/src/debug/debug_info.rs

```diff
@@ -747,22 +791,93 @@ impl DebugInfo {
             };
 
             // Part 1-b: Check LR values to determine if we can continue unwinding.
-            // TODO: ARM has special ranges of LR addresses to indicate fault conditions. We should check those also.
             if let Some(check_return_address) = unwind_registers.get_return_address() {
                 if check_return_address.is_max_value() || check_return_address.is_zero() {
                     // When we encounter the starting (after reset) return address, we've reached the bottom of the stack, so no more unwinding after this.
                     stack_frames.push(return_frame);
                     tracing::trace!("UNWIND: Stack unwind complete - Reached the 'Reset' value of the LR register.");
                     break;
+                } else {
+                    // Part 1-c: If we just processed an exception handler,  we need to update the `unwind_registers` to match the frame that invoked the exception handler.
+                    if exception_return_context != ExceptionReturnContext::NoException {
+                        tracing::trace!(
+                            "UNWIND: Stack unwind reached an exception handler {exception_return_context:?}",
+                        );
+                        match core.core_type() {
+                            CoreType::Armv6m | CoreType::Armv7em | CoreType::Armv7m => {
```

Comment:
Why not Armv8m here? Can help with testing if necessary.

Reply:
https://github.com/probe-rs/probe-rs/pull/1495#discussion_r1099174952